### PR TITLE
fix(cairo): enforce correct table and columns names on all operations

### PR DESCRIPTION
**Pull request by ideoma**
Security fixes:

- verify table names when created from ILP
- verify column names
- do not allow control characters in table / column names
- `TableWriter` to not delete directories inside the table directory which don't match partition name pattern
- check ILP auth signature to not be all zeros

Test fixes:
- regenerated test ILP Auth key to be compatible after JDK 18.0.1.1 release

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 74 / 75 (98.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|10|11|90.91%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/std/str/AbstractCharSink.java|12|12|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|30|30|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContext.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoException.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cairo/TableUtils.java [hunk 1]
```diff
@@ -432,33 +476,51 @@ public static LPSZ iFile(Path path, CharSequence columnName, long columnTxn) {
...
-        for (int i = 0, l = seq.length(); i < l; i++) {
-            char c = seq.charAt(i);
+    public static boolean isValidTableName(CharSequence tableName) {
+        int l = tableName.length();
+        for (int i = 0; i < l; i++) {
+            char c = tableName.charAt(i);
             switch (c) {
-                case '?':
                 case '.':
+                    if (i == 0 || i == l - 1 || tableName.charAt(i - 1) == '.') {
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/cairo/TableUtils.java:485]: would be good to add some explanation for these rules. For example, I didn't know Windows does not allow a filename ending with dot.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 74 / 75 (98.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|10|11|90.91%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/std/str/AbstractCharSink.java|12|12|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|30|30|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContext.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoException.java|2|2|100.00%|

File: core/src/main/java/io/questdb/PropertyKey.java [hunk 2]
```diff
@@ -351,7 +351,8 @@ public enum PropertyKey {
     CAIRO_SQL_COLUMN_PURGE_RETRY_DELAY("cairo.sql.column.purge.retry.delay"),
     CAIRO_SQL_COLUMN_PURGE_RETRY_DELAY_MULTIPLIER("cairo.sql.column.purge.retry.delay.multiplier"),
     CAIRO_SQL_COLUMN_PURGE_RETRY_LIMIT_DAYS("cairo.sql.column.purge.retry.limit.days"),
-    CAIRO_SQL_SYSTEM_TABLE_PREFIX("cairo.system.table.prefix");
+    CAIRO_SQL_SYSTEM_TABLE_PREFIX("cairo.system.table.prefix"),
+    CAIRO_MAX_FILE_NAME_LENGTH("cairo.max.file.name.length");
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/PropertyKey.java:355]: what's the reason for introducing a new property?

File: core/src/main/java/io/questdb/cairo/TableUtils.java [hunk 3]
```diff
@@ -432,33 +481,60 @@ public static LPSZ iFile(Path path, CharSequence columnName, long columnTxn) {
         return path.$();
     }
 
-    public static boolean isValidColumnName(CharSequence seq) {
-        for (int i = 0, l = seq.length(); i < l; i++) {
-            char c = seq.charAt(i);
+    public static boolean isValidTableName(CharSequence tableName, int fsFileNameLimit) {
+        int l = tableName.length();
+        if (l > fsFileNameLimit) {
+            // Most file systems don't support files
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/cairo/TableUtils.java:487]: comment cut in the midde?

Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/TableUtils.java]: changed in https://github.com/questdb/questdb/pull/2172/commits/02bc931e7f7ceb0bc15adbb2e09865f152366a01

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 99 / 100 (99.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|10|11|90.91%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/DefaultLineUdpReceiverConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/str/AbstractCharSink.java|12|12|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|30|30|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/LineUdpParserImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContext.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoException.java|2|2|100.00%|

**Pull request merged**